lib/thread.c

```diff
@@ -378,13 +384,14 @@ thread_master_create (void)
   rv->timer->update = rv->background->update = thread_timer_update;
   rv->spin = true;
   rv->handle_signals = true;
+  pipe (rv->io_pipe);
+  fcntl (rv->io_pipe[0], F_SETFL, fcntl(rv->io_pipe[0], F_GETFL, 0) | O_NONBLOCK);
```

Comment:
There's a function in `lib/network.c` called `set_nonblocking` for that, maybe you want to use it.